src/libs/routing/routes/explorer.ts

```diff
@@ -0,0 +1,68 @@
+import { Route, redirect } from '@tanstack/react-router';
+import { ExplorerType } from 'components/explorer/utils';
+import { rootRoute } from 'libs/routing/routes/root';
+import { ExplorerPage } from 'pages/explorer';
+import { ExplorerTypePage } from 'pages/explorer/type';
+import { ExplorerTypeOverviewPage } from 'pages/explorer/type/overview';
+import { ExplorerTypePortfolioPage } from 'pages/explorer/type/portfolio';
+import { ExplorerTypePortfolioTokenPage } from 'pages/explorer/type/portfolio/token';
+
+export const explorerLayout = new Route({
+  getParentRoute: () => rootRoute,
+  path: '/explorer',
+});
+
+export const explorerRedirect = new Route({
+  getParentRoute: () => explorerLayout,
+  path: '/',
+  beforeLoad: () => {
+    redirect({
+      to: '/explorer/$type',
+      params: { type: 'token-pair' },
+      throw: true,
+    });
+  },
+});
+
+export const explorerPage = new Route({
+  getParentRoute: () => explorerLayout,
+  path: '$type',
+  parseParams: (params: Record<string, string>) => {
+    return { type: params.type as ExplorerType };
+  },
```

Comment:
couldn't be rewritten like that ? 
```suggestion
  parseParams: <T extends { type:  ExplorerType}>({ type }: T) => { type },
```

Reply:
this doesn't seem to work unfortunately. 
![telegram-cloud-photo-size-1-4994821896533028042-y](https://github.com/bancorprotocol/carbon-app/assets/7262066/533beeaf-1ceb-4742-b426-7c719f012658)